docs/product-user-guide/src/main/asciidoc/decision-tables-defining-proc.adoc

```diff
@@ -0,0 +1,51 @@
+
+*<@Michael (1 of 5): I've added the above note both here and in the "Uploading decision tables" section to clarify that a single spreadsheet should be used. That work?>*
+
+Refer to the following sample spreadsheet as you define your decision table:
+
+.Sample decision table spreadsheet
+image::1248.jpg[]
+
+.Procedure
+. In a new XLS or XLSX spreadsheet, go to the second or third column and label a cell `RuleSet` (row 9 in example). Reserve the column or columns to the left for descriptive metadata (optional).
+. In the next cell to the right, enter a name for the `RuleSet`. This named rule set will contain all `RuleTable` rules defined in the rule package.
+. Under the `RuleSet` cell, define any rule attributes (one per cell) that you want to apply globally to all rule tables in the package. Specify attribute values in the cells to the right. For supported attributes, see xref:decision-tables-rule-set-entries-ref[].
+. Below the the `RuleSet` area and in the same column as the `RuleSet` cell, skip a row and label a new cell `RuleTable` (row 13 in example) and enter a table name in the same cell. The name string is used as the initial part of the name for all rules derived from this rule table, with the row number appended for distinction. You can override this automatic naming by inserting a `NAME` attribute column.
+. Use the next four rows to define the following elements as needed (rows 14-17 in example):
++
+* *Rule attributes:* Conditions, actions, or other attributes. For supported attributes, see xref:decision-tables-rule-table-entries-ref[].
+* *Object types:* The data objects to which the rule attributes apply. If the same object type applies to multiple columns, merge the object cells into one cell across multiple columns, instead of repeating the object type in multiple cells. When an object type is merged, all columns below the merged range will be combined into one set of constraints within a single pattern for matching a single fact at a time. When an object is repeated in separate columns, the separate columns can create different patterns, potentially matching different or identical facts.
+* *Constraints:* Constraints on the object types.
+* *Column label:* (Optional) Any descriptive label for the column, as a visual aid. Leave blank if unused.
++
+NOTE: As an alternative to populating both the object type and constraint cells, you can enter the full expression in either cell and leave the other blank. For example, instead of `Person` as the object type and `age` as the constraint (separate cells), you can enter `Person( age == $1)` in the object type cell and leave the constraint empty, or vice versa.
+
++
+*<@Michael (2 of 5): I've added the above note to clarify that the full expression can be populated in either object type or constraint cells. That work?>*
```

Comment:
@sterobin Not entirely true (and I think my last comment was probably misleading); if you have a value in the "pattern" cell (first row under CONDITION) it cannot contain a "code snippet" (e.g. ```Person( age == $param )```) as it gives a compilation error. You'd need ```Person``` in the "pattern" cell and ```age == $param``` (or simply ```age```) in the "constraint" cell (second row under CONDITION). However you can leave the "pattern" cell empty any put ```Person( age == $param )``` in ONLY the "constraint" cell. 

So, I think in summary the original over-use of "optional" should be:

- Pattern cell: Optional (IF Constraint cell is a full expression) otherwise optional.
- Constraint cell: Compulsory.